js_modules/dagit/packages/core/src/assets/AssetPartitions.test.tsx

```diff
@@ -0,0 +1,155 @@
+import {MockedProvider} from '@apollo/client/testing';
+import {act, render, screen, waitFor} from '@testing-library/react';
+import userEvent, {specialChars} from '@testing-library/user-event';
+import React from 'react';
+import {Route} from 'react-router-dom';
+
+import {AssetKeyInput} from '../graphql/types';
+import {TestProvider} from '../testing/TestProvider';
+
+import {AssetPartitionListProps} from './AssetPartitionList';
+import {AssetPartitions} from './AssetPartitions';
+import {AssetViewParams} from './AssetView';
+import {
+  SingleDimensionStaticPartitionHealthQuery,
+  SingleDimensionTimePartitionHealthQuery,
+} from './PartitionHealthSummary.mocks';
+
+// This file must be mocked because useVirtualizer tries to create a ResizeObserver,
+// and the component tree fails to mount. We still want to test whether certain partitions
+// are shown, so we print the keys in a simple "list".
+jest.mock('./AssetPartitionList', () => ({
+  AssetPartitionList: (props: AssetPartitionListProps) => (
+    <div>
+      <div data-testid="focused-partition">{props.focusedDimensionKey}</div>
+      {props.partitions.slice(0, 50).map((p) => (
+        <div key={p} onClick={() => props.setFocusedDimensionKey?.(p)}>
+          {p}
+        </div>
+      ))}
+    </div>
+  ),
+}));
+
+const SingleDimensionAssetPartitions: React.FC<{assetKey: AssetKeyInput}> = ({assetKey}) => {
+  const [params, setParams] = React.useState<AssetViewParams>({});
+  return (
+    <TestProvider>
```

Comment:
Is the full `TestProvider` definitely needed for your tests? I've been trying to avoid it where possible when using `MockedProvider`, since `TestProvider` includes the full Apollo mocked schema and everything else. No big deal either way though.

Reply:
Ahh I think I can do without, will change these! I think it's really just the workspace context that ends up weirdly required in these.